src/app/shared/filter-bar/filter-bar.component.ts

```diff
@@ -55,11 +70,22 @@ export class FilterBarComponent implements OnInit, OnDestroy {
     });
 
     this.views$.subscribe(() => this.applyFilter());
+
+    this.breakpointSubscription = this.breakpointObserver.observe([Breakpoints.Small, Breakpoints.XSmall]).subscribe((result) => {
+      this.isSmallScreen = result.matches;
+      this.updateSpan();
+    });
+
+    this.breakpointSubscription = this.breakpointObserver.observe([Breakpoints.Medium]).subscribe((result) => {
+      this.isMediumScreen = result.matches;
+      this.updateSpan();
+    });
```

Comment:
While this works, would it be better to take advantage of the auto scaling capabilities of css? 
We can avoid these breakpoints and corresponding values since the parent colspan of 7 scales according to screen size already, such that the children widths are always relative to the parent without changing their individual colspan or the parents. 
In order to preserve the full child item being shown when the screen shrinks, consider instead using min-width/max-width css properties with media queries to shift responsiveness logic to css where possible.